### PR TITLE
Add initial SPIFFE Workload API client

### PR DESCRIPTION
**Pull request by amartinezfayo**
This is a initial commit to have a SPIFFE Workload API client int the c-spiffe library.
Note that tests are pending.

### CODE REVIEW

File: client.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
+ * This is a sample client that uses the spiffe::WorkloadAPIClient class
+ * to fetch X.509 SVIDs
+ */
+
+#include <unistd.h>
+#include <csignal>
+#include "c-spiffe.h"
+
+#define INITIAL_DELAY 1000000
```
Comment (MarcosDY) [inline client.cc]: to make it easy to read, can you add a comment with the time unit used here?

File: client.cc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * Fetch SVIDs. If there is a failure, it retries with back-off.
+ */
+void fetchX509SVIDs() {
+    spiffe::WorkloadAPIClient workloadClient("unix:/tmp/agent.sock", updatedX509Callback);
+
+    Start:
+    workloadClient.FetchX509SVIDs();
+    if (workloadClient.GetFetchX509SVIDsStatus().ok()) {
+        std::cout << "FetchX509SVID rpc succeeded." << std::endl;
+    } else {
```
Comment (MarcosDY) [inline client.cc]: else is no necessary you can: 
```    
   if (!workloadClient.GetFetchX509SVIDsStatus().ok()) {
        std::cout << "FetchX509SVID rpc failed. Error code: " << workloadClient.GetFetchX509SVIDsStatus().error_code() << ". Error message: " << workloadClient.GetFetchX509SVIDsStatus().error_message() << std::endl;
        usleep(delay);
        if (delay >= TIMEOUT) {
             throw std::runtime_error("Timeout");
        }
       delay += delay;
        goto Start;           
   }
   std::cout << "FetchX509SVID rpc succeeded." << std::endl;
```

File: c-spiffe.cc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    X509SVIDResponse x509SVIDResponse;
+    ClientContext context;
+    context.AddMetadata("workload.spiffe.io", "true");
+
+    std::unique_ptr<ClientReader<X509SVIDResponse>> reader(stub->FetchX509SVID(&context, x509SVIDRequest));
+
+    while (reader->Read(&x509SVIDResponse)) {
+        updatedCallback(x509SVIDResponse);
+        if (stopFetchingX509SVIDs) {
+            return;
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply c-spiffe.cc]: Good catch, thanks!

**Comment by amartinezfayo** [comment]
@evan2645 yes, I thought about that too :). I explored the option to have a git submodule for that but discarded it when realized that it would require cloning the entire spire repo. I filed #3 to find a solution for this.

**Pull request merged**